app/Process.php

```diff
@@ -162,19 +185,50 @@ protected function run(string $command) : int
         return proc_close($proc);
     }
 
+    public function getOutputBuffer(): string
+    {
+        if (! $this->outputFileName) {
+            return '';
+        }
+
+        $output = file_get_contents($this->outputFileName);
+
+        if ($output === false) {
+            return 'Error: Unexpected failure trying to read the output file!';
+        }
+
+        return trim($output);
+    }
+
     protected function getDescriptors() : array
     {
         if ($this->output || env('FWD_VERBOSE')) {
             return [STDIN, STDOUT, STDERR];
         }
 
-        $devNull = fopen('/dev/null', 'w');
-
-        return [STDIN, $devNull, $devNull];
+        $outputfile = $this->outputFileName ?: '/dev/null';
+
+        return [
+            STDIN,
+            [
+                'file',
+                $outputfile,
+                'w',
+            ],
+            [
+                'file',
+                $outputfile,
+                'a',
```

Comment:
huh, I guess this must be `a` due to the fact that the file will already be opened with `w` above, and it would overwrite each others.

I suggest instead that we hold an open file descriptor ourselves under this class - that is we do the `fopen('', 'w+')`, and pass in as parameters here both of them, so we have a single resource writing to that file, and not two different file descriptors. Doing that we can store it like `$this->outputFd` or something like that, and make sure to close it in the `__destroy` or upon a new command execution.